autosubmit/platforms/paramiko_platform.py

```diff
@@ -1018,103 +1037,91 @@ def x11_status_checker(self, session, session_fileno):
-        streams are returned as Python ``file``-like objects representing
-        stdin, stdout, and stderr.
+        A new ``.Channel`` is open and the requested command is executed.
+        The command's input and output streams are returned as Python
+        ``file``-like objects representing stdin, stdout, and stderr.
 
         :param x11:
         :param retries:
-        :param get_pty:
         :param command: the command to execute.
         :type command: str
         :param bufsize: interpreted the same way as by the built-in ``file()`` function in Python.
         :type bufsize: int
-        :param timeout: set command's channel timeout. See `Channel.settimeout`.settimeout.
+        :param timeout: set command's channel timeout. See ``Channel.settimeout``.
         :type timeout: int
         :return: the stdin, stdout, and stderr of the executing command
-
-        :raises SSHException: if the server fails to execute the command
         """
         while retries > 0:
             try:
                 if x11:
-                    display = os.getenv('DISPLAY')
-                    if display is None or not display:
-                        display = "localhost:0"
+                    display = os.getenv('DISPLAY') or 'localhost:0'
                     try:
                         self.local_x11_display = xlib_connect.get_display(display)
                     except Exception as e:
-                        Log.warning(f"X11 display not found: {e}")
+                        Log.warning(f"X11 display not found: {str(e)}")
                         self.local_x11_display = None
                     chan = self.transport.open_session()
-                    chan.request_x11(single_connection=False,handler=self.x11_handler)
-                else:
-                    chan = self.transport.open_session()
-                if x11:
+                    chan.request_x11(single_connection=False, handler=self.x11_handler)
+
                     if "timeout" in command:
                         timeout_command = command.split("timeout ")[1].split(" ")[0]
                         if timeout_command == 0:
                             timeout_command = "infinity"
                         command = f'{command} ; sleep {timeout_command} 2>/dev/null'
-                    #command = f'export display {command}'
                     Log.info(command)
                     try:
                         chan.exec_command(command)
-                    except BaseException as e:
+                    except Exception as e:
                         raise AutosubmitCritical(f"Failed to execute command: {e}")
                     chan_fileno = chan.fileno()
                     self.poller.register(chan_fileno, select.POLLIN)
                     self.x11_status_checker(chan, chan_fileno)
                 else:
+                    chan = self.transport.open_session()
                     chan.exec_command(command)
+
```

Comment:
The changes above are mainly cosmetics. Worth mentioning: Catching `Exception` to avoid keyboard interrupt or system exit, etc. (recommended best practice, if we do have something from baseexception and not exception, there's probably not much we can do, and we can actually stall things by capturing it...)

And also the change in the logic above, before there were two `if x11/else`, now there's  single `if x11/else`.